apps/prairielearn/src/pages/enroll/enroll.ts

```diff
@@ -66,29 +70,38 @@ router.post(
         // Enrollment limits can only be configured on enterprise instances, so
         // we'll also only check and enforce the limits on enterprise instances.
         if (isEnterprise()) {
-          const {
-            institution,
-            course_instance,
-            institution_enrollment_count,
-            course_instance_enrollment_count,
-          } = await queryRow(
-            sql.select_and_lock_enrollment_counts,
+          const { course, course_instance } = await queryRow(
+            sql.select_course_instance,
             { course_instance_id: req.body.course_instance_id },
             z.object({
-              institution: InstitutionSchema,
+              course: CourseSchema,
               course_instance: CourseInstanceSchema,
-              institution_enrollment_count: z.number(),
-              course_instance_enrollment_count: z.number(),
             }),
           );
+          const institution = await queryRow(
+            sql.select_and_lock_institution,
+            { institution_id: course.institution_id },
+            InstitutionSchema,
+          );
+
+          const institutionEnrollmentCounts = await getEnrollmentCountsForInstitution({
+            institution_id: institution.id,
+            created_since: '1 year',
```

Comment:
This is directly using a Postgres interval - is that weird?

Reply:
Maybe? I think it's fine.

Reply:
Cool, I'll leave this as-is for now.